yt_dlp/extractor/vk.py

```diff
@@ -511,37 +511,22 @@ def _real_extract(self, url):
 class VKUserVideosIE(VKBaseIE):
     IE_NAME = 'vk:uservideos'
     IE_DESC = "VK - User's Videos"
-    _VALID_URL = r'https?://(?:(?:m|new)\.)?vk\.com/videos(?P<id>-?[0-9]+)(?!\?.*\bz=video)(?:[/?#&](?:.*?\bsection=(?P<section>\w+))?|$)'
+    _VALID_URL = r'https?://(?:(?:m|new)\.)?vk\.com/video/@(?P<id>[^?$#/&]+)(?!\?.*\bz=video)(?:[/?#&](?:.*?\bsection=(?P<section>\w+))?|$)'
     _TEMPLATE_URL = 'https://vk.com/videos'
     _TESTS = [{
-        'url': 'https://vk.com/videos-767561',
```

Comment:
Verify that the old URL redirects correctly via generic

Reply:
rip
it doesn't 🤦‍♂️ 
```
C:\Users\Ashish\Documents\code\Ashish0804\yt-dlp [vkuser ≡ +1 ~0 -0 !]> ptf https://vk.com/videos-767561
[debug] Command-line config: ['-F', '-v', '--list-subs', 'https://vk.com/videos-767561']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.12.27 [6223f67a8] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 1f2d5721c
[debug] Python version 3.8.8 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg N-105029-gf3a949dc4c-20211225 (setts), ffprobe N-105029-gf3a949dc4c-20211225
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://vk.com/videos-767561
[generic] videos-767561: Requesting header
WARNING: [generic] Could not send HEAD request to https://vk.com/videos-767561: HTTP Error 418:
[generic] videos-767561: Downloading webpage
WARNING: [generic] Falling back on generic information extractor.
[generic] videos-767561: Extracting information
[debug] Looking for video embeds
[redirect] Following redirect to https://vk.com/badbrowser.php
[debug] [generic] Extracting URL: https://vk.com/badbrowser.php
[generic] badbrowser: Requesting header
WARNING: [generic] Could not send HEAD request to https://vk.com/badbrowser.php: HTTP Error 418:
[generic] badbrowser: Downloading webpage
WARNING: [generic] Falling back on generic information extractor.
[generic] badbrowser: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://vk.com/badbrowser.php
Traceback (most recent call last):
  File "C:\Users\Ashish\Documents\code\Ashish0804\yt-dlp\yt_dlp\YoutubeDL.py", line 1365, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\Ashish\Documents\code\Ashish0804\yt-dlp\yt_dlp\YoutubeDL.py", line 1435, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\Ashish\Documents\code\Ashish0804\yt-dlp\yt_dlp\extractor\common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\Ashish\Documents\code\Ashish0804\yt-dlp\yt_dlp\extractor\generic.py", line 3880, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://vk.com/badbrowser.php
```
Tho i don't want to waste time writing OR statements in the regex and then checking if we got the old url or the url and then proceed.

So if u have to have backward compatibility with old urls then u can either make changes and push (i have allowed edits) or close this PR and maybe someone else can work on it

Reply:
This is actually caused by a bug in the generic extractor. The issue is not trivial to fix, so I'll leave it for later